src/java/fr/paris/lutece/plugins/suggest/business/SuggestSubmitTypeDAO.java

```diff
@@ -290,14 +290,9 @@ public boolean isAssociateToSuggest( int nIdType, Plugin plugin )
             daoUtil.setInt( 1, nIdType );
             daoUtil.executeQuery( );
     
-            if ( daoUtil.next( ) )
+            if ( daoUtil.next( ) && daoUtil.getInt( 1 ) != 0)
             {
-                if ( daoUtil.getInt( 1 ) != 0 )
-                {
-                    daoUtil.free( );
-    
-                    return true;
-                }
+                return true;
```

Comment:
Shouldn't you go ahead and just return `daoUtil.next( ) && daoUtil.getInt( 1 ) != 0` like other changes you did ?

Reply:
Yes, indeed.